src/sysfiles.c

```diff
@@ -865,26 +878,26 @@ Int SyFopen (
     if(strlen(mode) >= 2 && mode[1] == 'b')
        flags |= O_BINARY;
 #endif
+
     /* try to open the file                                                */
     if ( 0 <= (syBuf[fid].fp = open(name,flags, 0644)) ) {
-        syBuf[fid].pipe = 0;
+        syBuf[fid].type = raw_socket;
         syBuf[fid].echo = syBuf[fid].fp;
-        syBuf[fid].ateof = 0;
-        syBuf[fid].crlast = 0;
         syBuf[fid].bufno = -1;
-        syBuf[fid].isTTY = 0;
+    }
+    else if (strncmp(mode, "r", 1) == 0 && SyIsReadableFile(namegz) == 0 &&
+             (syBuf[fid].gzfp = gzopen(namegz, mode))) {
+        syBuf[fid].type = gzip_socket;
+        syBuf[fid].bufno = -1;
     }
 #ifdef HAVE_POPEN
    else if ( strncmp(mode,"r",1) == 0
            && SyIsReadableFile(namegz) == 0
              && ( (syBuf[fid].pipehandle = popen(cmd,"r"))
                ) ) {
-        syBuf[fid].pipe = 1;
-        syBuf[fid].fp = fileno(syBuf[fid].pipehandle);
-        syBuf[fid].ateof = 0;
-        syBuf[fid].crlast = 0;
-        syBuf[fid].bufno = -1;
-        syBuf[fid].isTTY = 0;
+       syBuf[fid].type = pipe_socket;
+       syBuf[fid].fp = fileno(syBuf[fid].pipehandle);
+       syBuf[fid].bufno = -1;
```

Comment:
Weirdly, the indentation in this case is wrong by one space (but again, I am not really asking you to change this)